sponsorship_compassion/models/contracts.py

```diff
@@ -537,8 +537,9 @@ def contract_active(self):
 
         # Update sponsorships on partner
         partners = self.mapped('partner_id') | self.mapped('correspondant_id')
-        partners._compute_has_sponsorships()
-
+        partners.number_sponsorships += 1
+        partners.has_sponsorships = partners.number_sponsorships
+        partners.update_church_sponsorships_number(True)

```

Comment:
You cannot update church in this module (must be done in compassion-switzerland)